### PR TITLE
Bugfixes and improvements for "fancy" storages

### PR DESCRIPTION
**Pull request by Amunak**
This PR fixes some minor things about "fancy" storages (implemented in #9331).

* Fixed a bug where you could open them from anywhere. Adjacency is now required.
* Refactored the icon update from the `AltClick` proc to `update_icon` for more consistent behavior. I also added parent proc calls for all overriden `update_icon` calls for this to work properly, which in turn meant some other minor changes in some of the overriden procs.
* Fixed vial packs being "openable" - forced them open and non-closable. Among other things this prevents the stupid sound and (after making the previous change) breaking the sprite.
* Fixed cigarette packs spawning in opened. They can now be toggled open/closed and behave nicely like match boxes.
* Since cigarette packs can now be opened and closed I made it impossible to pull a cig from a closed packet.
* Storage will be opened before inserting anything. When inserting you can even see the box open with the empty slot and then the item is inserted.

I have tested this fix fairly rigorously on all the "special" items that override `update_icon`, as well as on the more "regular" items like candle boxes. All seems to work as expected, the sprites update nicely.

### CODE REVIEW

**Comment by Amunak** [comment]
!bugfix (I think?)
!review

**Comment by Amunak** [comment]
!bugfix

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 1]
```diff
@@ -163,6 +175,7 @@
 	starts_with = null
 
 /obj/item/storage/box/fancy/crayons/update_icon()
+	.=..()
```
Comment (fernerr) [inline code/game/objects/items/weapons/storage/fancy.dm:178]:
```suggestion
	. = ..()
```
Just to keep the style consistent.

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 2]
```diff
@@ -255,17 +268,22 @@
 		new cigarette_to_spawn(src)
 
 /obj/item/storage/box/fancy/cigarettes/update_icon()
-	icon_state = "[initial(icon_state)][contents.len]"
+	.=..()
```
Comment (fernerr) [inline code/game/objects/items/weapons/storage/fancy.dm:271]:
```suggestion
	. = ..()
```
Ditto

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 3]
```diff
@@ -378,6 +398,7 @@
 	queue_icon_update()
 
 /obj/item/storage/lockbox/vials/update_icon(var/itemremoved = 0)
+	.=..()
```
Comment (fernerr) [inline code/game/objects/items/weapons/storage/fancy.dm:401]:
```suggestion
	. = ..()
```

**Review by fernerr — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**